### PR TITLE
Kotlin dsl kotlin conventions

### PR DESCRIPTION
**Pull request by qwwdfsad**
#2474 that preserves git history

### CODE REVIEW

File: buildSrc/src/main/kotlin/kotlin-jvm-conventions.gradle.kts [hunk 1]
```diff
@@ -4,37 +4,43 @@
...
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_6
+    targetCompatibility = JavaVersion.VERSION_1_6
+}
 
-if (rootProject.ext.jvm_ir_enabled) {
-    kotlin.target.compilations.all {
+if (rootProject.extra.get("jvm_ir_enabled") as Boolean) {
+    kotlin.target.compilations.configureEach {
```
Comment (qwwdfsad) [inline buildSrc/src/main/kotlin/kotlin-jvm-conventions.gradle.kts:19]: @turansky please elaborate on using `configureEach` instead of `all`

Comment (turansky) [reply buildSrc/src/main/kotlin/kotlin-jvm-conventions.gradle.kts:19]: This change created according [Gradle migration guide](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:old_vs_new_configuration_api_overview)

Comment (turansky) [reply buildSrc/src/main/kotlin/kotlin-jvm-conventions.gradle.kts:19]: Screen Shot 2021-01-18 at 7 06 19 PM

Comment (qwwdfsad) [reply buildSrc/src/main/kotlin/kotlin-jvm-conventions.gradle.kts:19]: Okay, thanks!

**Pull request merged**